client/state/initial-state.js

```diff
@@ -24,7 +25,8 @@ export const MAX_AGE = 7 * DAY_IN_HOURS * HOUR_IN_MS;
 function getInitialServerState() {
 	// Bootstrapped state from a server-render
 	if ( typeof window === 'object' && window.initialReduxState ) {
-		return window.initialReduxState;
+		const serverState = reducer( window.initialReduxState, { type: SERVER_DESERIALIZE } );
+		return pick( serverState, Object.keys( window.initialReduxState ) );
```

Comment:
Currently the state tree being bootstrapped is defined by the [`pick` on the server](https://github.com/Automattic/wp-calypso/blob/bcfe9f49d801818e56bfa87ea8dce5709dff88b9/server/pages/index.js#L382). This works fine for now, but perhaps once the server code becomes more modularized, we can remove the server `pick` and change this client code to use a formally-defined bootstrap tree.

A reducer that did not return any default state for the `SERVER_DESERIALIZE` action type would be great for that. Not sure if that is possible.